packages/core/src/execution-engine/node-execution-context/utils/parse-request-object.ts

```diff
@@ -0,0 +1,467 @@
+	keys.forEach((key) => {
+		const formField = data[key];
+
+		if (formField instanceof Array) {
+			formField.forEach((item) => {
+				pushFormDataValue(formData, key, item);
+			});
+		} else {
+			pushFormDataValue(formData, key, formField);
+		}
+	});
+	return formData;
+};
+
+function searchForHeader(config: AxiosRequestConfig, headerName: string) {
+	if (config.headers === undefined) {
+		return undefined;
+	}
+
+	const headerNames = Object.keys(config.headers);
+	headerName = headerName.toLowerCase();
+	return headerNames.find((thisHeader) => thisHeader.toLowerCase() === headerName);
+}
+
+async function generateContentLengthHeader(config: AxiosRequestConfig) {
+	if (!(config.data instanceof FormData)) {
+		return;
+	}
+	try {
+		const length = await new Promise<number>((res, rej) => {
+			// eslint-disable-next-line @typescript-eslint/no-unsafe-call, @typescript-eslint/no-unsafe-member-access
+			config.data.getLength((error: Error | null, dataLength: number) => {
+				if (error) rej(error);
+				else res(dataLength);
+			});
+		});
+		config.headers = {
+			...config.headers,
+			'content-length': length,
+		};
+	} catch (error) {
+		// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+		Container.get(Logger).error('Unable to calculate form data length', { error });
+	}
+}
+
+const getHostFromRequestObject = (
+	requestObject: Partial<{
+		url: string;
+		uri: string;
+		baseURL: string;
+	}>,
+): string | null => {
+	try {
+		const url = (requestObject.url ?? requestObject.uri) as string;
+		return new URL(url, requestObject.baseURL).hostname;
+	} catch (error) {
+		return null;
+	}
+};
+
+const getBeforeRedirectFn =
+	(agentOptions: AgentOptions, axiosConfig: AxiosRequestConfig) =>
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	(redirectedRequest: Record<string, any>) => {
+		const redirectAgent = new Agent({
+			...agentOptions,
+			// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+			servername: redirectedRequest.hostname,
+		});
+		redirectedRequest.agent = redirectAgent;
+		// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+		redirectedRequest.agents.https = redirectAgent;
+
+		if (axiosConfig.headers?.Authorization) {
+			// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
+			redirectedRequest.headers.Authorization = axiosConfig.headers.Authorization;
+		}
+		if (axiosConfig.auth) {
+			redirectedRequest.auth = `${axiosConfig.auth.username}:${axiosConfig.auth.password}`;
+		}
+	};
+
+// eslint-disable-next-line complexity
+export async function parseRequestObject(requestObject: IRequestOptions) {
+	// This function is a temporary implementation
+	// That translates all http requests done via
+	// the request library to axios directly
+	// We are not using n8n's interface as it would
+	// an unnecessary step, considering the `request`
+	// helper can be deprecated and removed.
```

Comment:
Can we move this comment into a JSDoc comment for this function, so it's more discoverable